<!DOCTYPE meta PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.or
g/TR/html4/loose.dtd">
<html>
<head>
<title>
GenePattern - STAR.aligner documentation
</title>
<style type="text/css"><!--
  h1 {text-align: center}
  span.terminal {font-family: Courier}
--></style>
</head>

<body>
<h1>STAR 2.5.3a aligner</h1>
<h2>Summary</h2>
STAR (Spliced Transcripts Alignment to a Reference) is a fast RNA-seq reads to genome mapper. It differs from other mappers as TopHat in that it gains speed at the expense of consuming more RAM and that it incorporates transcriptome annotation at the indexing build stage rather than at the analysis stage. It can optionally detect non-canonical splices and chimeric transcripts.

<h2>Description</h2>
STAR starts by searching for sequence substrings that match perfectly between a read and the genome. It can do this very fast by using an uncompressed suffix array. Then it tries to extend these substrings into "seed" alignements, allowing for mismatches and gaps. Finally, it tries to stitch the "seeds" together into mappings, taking into account introns and eventually mate pairs and chimeric transcripts.
<p>
STAR always first maps the reads to the genome. If splice junction annotation is available it is only later used to decide which potential splice junctions to accept, the criteria being more lax for annotated junctions.

<h3 id="two_pass">Two-pass alignment</h3>
STAR can be run in 2-pass mode. The 1<sup>st</sup> pass serves to detect novel junctions, and in the 2<sup>nd</sup> pass, the detected junctions are added to the annotated junctions, and all reads are re-mapped to finalize the alignments. While this procedure does not significantly increase the number of novel collapsed junctions, it substantially increases the number of reads crossing the novel junctions, by allowing novel splices with shorter overhang. This procedure is especially advantageous in cases where annotations are unavailable or incomplete. There are two ways to do it :
<ol>
<li>in a single run, by putting STAR.aligner is two-pass mode
<li>in 3 steps :
  <ol>
  <li>run STAR.aligner
  <li>run STAR.indexer, using the file <span class="terminal">&lt;basename&gt;.SJ.out.tab</span> of the first run as input. You can merge tab files from runs performed on different samples. 
  <li>run again STAR.aligner, using the index generated by STAR.indexer as input
  </ol>
</ol>
The second method has the advantage that the indexing must be done only once and that you can later run STAR.aligner as much as you like without the overhead of the on-the-fly merging of the extra annotation into the index. Also, the first method is slighty less sensitive because if a novel junction is highly expressed in only one sample and weakly (only a few reads with short overhang) in other samples, the per-sample 2-pass approach may only detect this junction in the former sample. On the other hand, the multi-sample 2-pass strategy will detect this junction in all samples.

<h3>Shared memory</h3>
STAR in the V.I.B. GenePattern server has been set to using shared memory. That means that if several runs are started on the same genome they use the same instance of the index loaded into "live" RAM memory. The memory is cleared after all runs are terminated. This saves on memory space as well as on overhead to load the index into memory.
<p>
This feature is disabled if STAR.aligner is run in two-pass mode since the run needs a personal version of the index to merge the extra annotation into. It is also disabled for generating sorted BAM output (and for generating "wiggle" output) since the sorting needs an unpredictable amount of extra memory.

<h2>Parameters</h2>
<table border><tr>
  <th>Name</th>
  <th>Description</th>
  <th>Allowed values</th>
  <th>Default</th>
</tr><tr>
  <th colspan="4">input</th>
</tr><tr>
  <td>star index</td>
  <td>A STAR index. Select a prebuilt index or upload your own as a ZIP file (you can make one using STAR.indexer).</td>
  <td>selection from dynamic list or valid file</td>
  <td></td>
</tr><tr>
  <td>reads pair 1</td>
  <td>Unpaired reads or first mate for paired reads, as files in fastA or fastQ format. You can provide several files.</td>
  <td>valid file(s)<br><b>(is required)</b></td>
</tr><tr>
  <td>reads pair 2</td>
  <td>Second mate for paired reads.<p><b><i>The files and the sequences inside the files must be in the same order as for the first mates.</b></i></td>
  <td>valid file(s)</b></td>
  <td></td>
</tr><tr>
  <th colspan="4">mapping and reporting of mapped reads</th>
</tr><tr>
  <td>max reads to align</td>
  <td>Set this if you want to map only a selected set of reads at the top of the input (mainly useful for testing).</td>
  <td>min = 1</td>
  <td></td>
</tr><tr>
  <td>align read end to end</td>
  <td>Align reads end to end and count all mistmatches. The default is to use instead "soft clipping" at both 5'- and 3'-end, that means to make a local alignment and ignore the contribution of the ends of the read to the final score if this improves the score. <b><i>This can be useful for ChIP-seq and other DNA sequencing applications with reads that have already been quality-trimmed.</i></b></td>
  <td><li>yes<li>no</td>
  <td>no</td>
</tr><tr>
  <td>max number mismatches</td>
  <td>Maximum number of mismatches per read. Note that a mate pair is counted as one read.</td>
  <td>min = 0</td>
  <td>10</td>
</tr><tr>
  <td>max fraction mistaches</td>
  <td>Maximum number of mismatches per read, expressed as the proportion between the number of mismatches and the <b><i>mapped</i></b> length of the read. Note that a mate pair is counted as one read.</td>
  <td>min = 0<br>max = 1</td>
  <td>0.3</td>
</tr><tr>
  <td>min overhang annotated read</td>
  <td>Minimum length that read must map at both sides of splice junction in order to accept mapping to a splice junction that is annotated in GTF of tab file.</td>
  <td>min = 1</td>
  <td>3</td>
</tr><tr>
  <td>min overhang not annotated read</td>
  <td>Minimum length that read must map at both sides of splice junction in order to accept de novo discovery of splice junction.</td>
  <td>min = 1</td>
  <td>5</td>
</tr><tr>
  <td>min intron length</td>
  <td>Minimum size of intron. A gap in the alignment between a read and the genome that is smaller is considered a deletion, not an intron.</td>
  <td>min = 1</td>
  <td>21</td>
</tr><tr>
  <td>max intron length</td>
  <td>Maximum size of intron. If a read aligns to the genome with a gap larger than this it is considered a chimeric read. <b><i>The default value of 500,000 is fine-tuned to mammalian genomes, for plant and yeast genomes you will have to decrease it.</i></b></td>
  <td>min = 1</td>
  <td>500000</td>
</tr><tr>
  <td>mates max gap</td>
  <td>Maximum distance between mate pair reads. If reads map to the genome farther apart the fragment is considered to be chimeric. <b><i>The default value of 500,000 is fine-tuned to mammalian genomes, for plant and yeast genomes you will have to decrease it.</i></b></td>
  <td>min = 1</td>
  <td>500000</td>
</tr><tr>
  <td>secondary mapping mismatches range</td>
  <td>By default STAR only reports reads that map to multiple locations on the genome when they map with the highest possible score. You can ask STAR to report also secondary mappings with up to this much more mismatches than the primary mappings.</td>
  <td>min = 0</td>
  <td>0</td>
</tr><tr>
  <td>max multimapping</td>
  <td>Do not report reads that map to more than that many different locations on the genome.</td>
  <td>min = 1</td>
  <td>10</td>
</tr><tr>
  <td>min report canonical junction overhang</td>
  <td>Criterium for reporting de novo predicted splice junctions in the SJ.tab.out file : for canonical splice sites at least one read with at least this overhang is needed.</td>
  <td>min = 1</td>
  <td>12</td>
</tr><tr>
  <td>min report noncanonical junction overhang</td>
  <td>Criterium for reporting de novo predicted splice junctions in the SJ.tab.out file : for noncanonical splice sites at least 3 reads with at least this overhang are needed.</td>
  <td>min = 1</td>
  <td>30</td>
</tr><tr>
  <td>map only reported juntions</td>
  <td>By default all mapped reads are output in the SAM/BAM file. Set this to yes if you want to bring the SAM/BAM file in agreement with the SJ.tab.out file, by outputting only reads that are not spliced or map to splice junctions that are annotated or have been de novo predicted.</td>
  <td><ul><li>yes<li>no</ul></td>
  <td>no</td>
</tr><tr>
  <th colspan="4">postprocessing and supplementary output</th>
</tr><tr>
  <td>tow pass</td>
  <td>Run STAR in 2-pass mode, that is, run STAR a first time, merge the found splice junctions with the splice junction annotation in the index, and run STAR a second time. Consult the <a href="#two_pass">Description section</a> for more explanation.</td>
  <td><ul><li>yes<li>no</ul></td>
  <td>no</td>
</tr><tr>
  <td>detect chimeric transcripts</td>
  <td></td>
  <td><ul><li>yes<li>no</ul></td>
  <td>no</td>
</tr><tr>
  <td>output unmapped reads</td>
  <td></td>
  <td><ul><li>yes<li>no</ul></td>
  <td>no</td>
</tr><tr>
  <td>quantify genes</td>
  <td>Write a table with number of reads mapped per gene and a supplementary BAM file with mappings to transcriptome instead of to genome coordinates.</td>
  <td><ul><li>yes<li>no</ul></td>
  <td>no</td>
</tr><tr>
  <td>output wiggle file</td>
  <td>utput a "wiggle" file for viewing in viewers like IGV or UCSC genome browser.</td>
  <td><ul><li>none<li>file in BedGraph format<li>file in WIG format</ul></td>
  <td>none</td>
</tr><tr>
  <td>wiggle signal</td>
  <td>From which bases to generate the signal for the "wiggle" file. By default they are all used, but you can choose to use only the bases at the 5'-end of the 1<sup>st</sup> read (useful for CAGE/RAMPAGE) or only from the 2<sup>nd</sup> mate from read pairs.</td>
  <td><ul><li>all<li>only 5' from read 1<li>only from read 2</ul></td>
  <td>all</td>
</tr><tr>
  <th colspan="4">output</th>
</tr><tr>
  <td>output format</td>
  <td></td>
  <td><ul><li>SAM unsorted<li>BAM unsorted<li>BAM sorted by coordinate</ul></td>
  <td>SAM unsorted</td>
</tr><tr>
  <td>HI flag</td>
  <td>Number of "hits" with higest score to label as "primary" in the SAM/BAM file. If you choose "only one" the others are labeled as "secondary" (all reads that map with lower than the highest score are labeled "secondary").</td>
  <td><ul><li>only one<li>all</ul></td>
  <td>only one</td>
</tr><tr>
  <td>output prefix</td>
  <td>The prefix to use for the output file names.</td>
  <td></td>
  <td>STAR</td>
</tr></table>

<h2>Input files</h2>
STAR.aligner takes as input a set of reads and a genome.
<p>
The reads can be provided in <a href="http://wiki.bits.vib.be/index.php/.fastq" target="_blank">fastQ</a> format or  in <a href="http://wiki.bits.vib.be/index.php/.fasta" target="_blank">fastA</a> format. The reads can be spread over several files.
<p>
It is possible to provide several files with reads. For mate pair experiments it is necessary to provide two sets of input files and it is important that the files as well as the reads inside each file are in the right order so that STAR can find the corresponding partners of each mate pair.
<p>
The genome must be provided as an index, there is no need for the original sequences and the original annotation. STAR.aligner has access to a series of prebuilt indexes. Alternatively, the user can provide an index of his own. The index must be in a ZIP file. GenePattern has a tool <b><i>STAR.indexer</i></b> to make the index from a series of fastA files.

<h2>Output files</h2>
STAR.aligner creates a whole series of output files. At minimum, a run will produce the following :
<ul>
<li><b>&lt;basename&gt;.Log.out</b> : an overview of the parametrization of the program
<li><b>&lt;basename&gt;.Log.final.out</b> : a final overview of the mapping process (unique reads, multi-mapping reads, unmapped reads, chimeric reads)
<li><b>&lt;basename&gt;.Log.progress.out</b> : a minute-by-minute follow-up of the mapping process (number of reads mapped, mapping rate,...)
<li><b>&lt;basename&gt;.Aligned.out.sam</b> : The main output, a file in <a href="http://wiki.bits.vib.be/index.php/.sam" target="_blank">SAM</a> format with the read to genome mappings. It contains as optional fileds :
  <ul>
  <li><span class="terminal">NH</span> : number of loci a reads maps to
  <li><span class="terminal">HI</span> : index for multimapping alignments
  <li><span class="terminal">AS</span> : alignment score
  <li><span class="terminal">nM</span> : number of mismatches
  </ul>
  You can choose to get instead a file <span class="terminal"> &lt;basename&gt;.Aligned.out.bam</span> in the more compact binary <a href="http://wiki.bits.vib.be/index.php/.bam" target="_blank">BAM</a> format and you can have it sorted as <span class="terminal"> &lt;basename&gt;.Aligned.sortedByCoord.out.bam</span>.
<li><b>&lt;basename&gt;.SJ.out.tab</b> : a file in tab-separated format with de novo predicted intron-exon splice junctions, consult the documentation of STAR.indexer for more explanation.
</ul>
<p>
By default STAR.aligner does not search for <b><i>fusion reads</i></b>, that are reads that map to regions of the genome that are located on different chromosomes, on different strands of the same chromosome or on widely distant regions of the same chromosome, so that they likely derive from sequencing <b><i>chimeric transcripts</i></b>. If you request it, STAR.aligner will output a separate file in SAM format <b>&lt;basename&gt;.Chimeric.out.sam</b>.
<p>
You can request STAR.aligner to write files <b>&lt;basename&gt;.Unmapped.out.mate1</b> (and eventually <b>&lt;basename&gt;.Unmapped.out.mate2</b>) with the reads that could not be mapped, in the same fastA or fastQ format as the input.
<p>
When you request STAR.aligner to quantify genes it will write 2 supplementary files :
<ul>
<li><b>&lt;basename&gt;.Aligned.toTranscriptome.out.bam</b> : a file in BAM format with mappings to the transcriptome (rather than to the genome as in the standard output file)
<li><b>&lt;basename&gt;.ReadsPerGene.out.tab</b> : a file in tab-separated format with on each line :
  <ol>
  <li>the gene ID
  <li>total number of reads mapped to the gene
  <li>number of reads mapped to the forward strand
  <li>number of reads mapped to the reverse strand
  </ol>
</ul>
<p>
You can request the output of "wiggle" files , which are useful for vizualization of the RNA-seq signal on genomic browsers as the <a href="http://genome.ucsc.edu/" target="_blank">UCSC genomic browser</a> or <a href="http://www.broadinstitute.org/igv" target="_blank">IGV</a>. The signal represents the number of reads crossing each genomic base. There are separate files for the two strands and there are separate files for uniquely and for multimappnig reads ; in the latter case the contribution of the multimappers will be divided by the number of loci they map to. You can choose between output in <a href="http://genome.ucsc.edu/goldenpath/help/bedgraph.html" target="_blank">BedGraph</a> or in <a href="http://genome.ucsc.edu/goldenpath/help/wiggle.html" target="_blank">WIG</a> format. STAR.aligner will write output files with respectively names <span class="terminal">&lt;basename&gt;.Signal.Unique.str1.out.bg</span>, <span class="terminal">&lt;basename&gt;.Signal.Unique.str2.out.bg</span>, <span class="terminal">&lt;basename&gt;.Signal.UniqueMultiple.str1.out.bg</span>, <span class="terminal">&lt;basename&gt;.Signal.UniqueMultiple.str2.out.bg</span> or <span class="terminal">&lt;basename&gt;.Signal.Unique.str1.out.wig</span>, <span class="terminal">&lt;basename&gt;.Signal.Unique.str2.out.wig</span>, <span class="terminal">&lt;basename&gt;.Signal.UniqueMultiple.str1.out.wig</span>, <span class="terminal">&lt;basename&gt;.Signal.UniqueMultiple.str2.out.wig</span>. Since the generation of the "wiggle" files demands reads sorted by coordinate, asking for "wiggle" output sets STAR.aligner automatically into making sorted BAM output.


<h2>Links</h2>
<ul>
<li>STAR at <a href="http://github.com/alexdobin/STAR"  target="_blank">GitHub</a>
</ul>

<h2>References</h2>
<ol>
<li>Dobin A, Davis CA, Schlesinger F, Drenkow J, Zaleski C, Jha S, Batut P, Chaisson M, Gingeras TR : STAR: ultrafast universal RNA-seq aligner. Bioinformatics 29(1):15-21 (2013). <a href="http://www.ncbi.nlm.nih.gov/pubmed/23104886" target="_blank">PubMed 23104886</a>
<li>Dobin A, Gingeras TR : Optimizing RNA-Seq Mapping with STAR. Methods Mol. Biol. 1415:245-265 (2016). <a href="http://www.ncbi.nlm.nih.gov/pubmed/27115637" target="_blank">PubMed 27115637</a>
</ol>

<h2>Author</h2>
The GenePattern interface is made by Guy Bottu, V.I.B.-B.I.T.S.
<p>
The STAR software is developed by a team of programmers headed by Alexander Dobin at Cold Spring Harbor Laboratory.

<h2>Version Comments</h2>
<table>
<tr><th>Version</th><th>Release date</th><th>Description</th><tr>
<tr><td>1</td><td>2016-08-29</td><td>for STAR 2.5.2a</td></tr>
</table>

</body>
</html>

