<!DOCTYPE meta PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.or
g/TR/html4/loose.dtd">
<html>
<head>
<title>
GenePattern - STAR.indexer documentation
</title>
<style type="text/css"><!--
  h1 {text-align: center}
  span.filename {font-family: Courier}
--></style>
</head>

<body>
<h1>STAR 2.5.3a indexer</h1>
<h2>Summary</h2>
The STAR indexer makes from a series of fastA files containing a genome an index for use with the STAR spliced read to genome aligner. It can optionally (but highly recommended !) incorporate information about known intron-exon junctions.

<h2>Parameters</h2>
<table border><tr>
  <th>Name</th>
  <th>Description</th>
  <th>Allowed values</th>
  <th>Default</th>
</tr><tr>
  <th colspan="4">input and output</th>
</tr><tr>
  <td>fasta file</td>
  <td>One or several fastA files containing the genome to index.</td>
  <td>valid file(s)<br><b>(is required)</b></td>
  <td></td>
</tr><tr>
  <td>GTF file</td>
  <td>A GTF file containing a list of gene model annotations. When provided, STAR will use it to improve the accuracy of the mapping of intron-exon splice junctions.</td>
  <td>selection from dynamic list or valid file</td>
  <td></td>
</tr><tr>
  <td>tab file</td>
  <td>A file in tab-separated format containing a list of gene model annotations, usually an SJ.tab.out file generated by a previous run of STAR.aligner. You can use it instead of or together with a GTF file. Consult the <a href="#tab_file">Input files</a> section for more explanation.</td>
  <td>valid file</td>
  <td></td>
</tr><tr>
  <td>index dir</td>
  <td>The name of the directory (and ZIP file) in which to put the STAR index.</td>
  <td>contiguous string of characters</td>
  <td></td>
</tr><tr>
  <th colspan="4">customization of indexing process</th>
</tr><tr>
  <td>overhang</td>
  <td>The length of the genomic sequence around the annotated intron-exon junctions. Ideally it should have a value <b>max(ReadLength) - 1</b>, but in most cases a generic value of 100 will work well.</td>
  <td>min = 1</td>
  <td>100</td>
</tr><tr>
  <td>index string length</td>
  <td>Length of the strings used for making the index. For small genomes (less than 1 Gbase) it can be useful to scale it down to <b>min(14, log<sub>2</sub>(GenomeLength)/2 - 1)</b>.</td>
  <td>min = 1</td>
  <td>14</td>
</tr><tr>
  <td>bin size</td>
  <td>Size of the bins for storage. If the genome contains a large number of separate scaffolds ("references") instead of just the full chromosomes and maybe a few scaffolds you can reduce RAM consumption by setting this to <b>min(18, log<sub>2</sub>(GenomeLength/NumberOfReferences))</b>.</td>
  <td>min = 1</td>
  <td>18</td>
</tr><tr>
  <td>RAM limit</td>
  <td>Upper limit for RAM allocated to indexing process, in Gbyte. For very large genomes you might heve to increase this, but make sure that the RAM is indeed physically present on the server and you have permission to make it available for the process. </td>
  <td>min = 1</td>
  <td>31</td>
</tr></table>

<h2>Input files</h2>
STAR.indexer takes as input one or several files in <a href="http://wiki.bits.vib.be/index.php/.fasta" target="_blank">fastA</a> format, containing the DNA sequences of a genome.
<p>
It can optionally take as supplementary input information about known intron-exon junctions. There are two ways to provide this information :
<ul>
<li>a file in <a href="http://wiki.bits.vib.be/index.php/.gtf" target="_blank">GTF</a> or <a href="http://wiki.bits.vib.be/index.php/.gff" target="_blank">GFF3</a> format. There should be lines with on the 3<sup>th</sup> field the word "<b>exon</b>", since these are the lines STAR uses to find where the intron-exon boundaries are. It is desirable that the same lines contain in their 9<sup>th</sup> field references "<b>transcript_id"</b> and "<b>gene_id</b>". It is important that the names of the chromosomes/scaffolds in the 1<sup>st</sup> field do match the names in the fastA genome files (e.g. <span class="filename">chr1</span> is not the same as <span class="filename">1</span>).
<li id="tab_file">a file in a tab-separated format native to the STAR package. Each line in the file must have 9 fields, e.g.
<pre>chr1	168166	169048	2	2	1	5	3	46</pre>
where the meaning of the fields is :
<ol>
  <li>the name of the chromosome/scaffold
  <li>the coordinate of the first base of the intron
  <li>the coordinate of the last base of the intron
  <li>strand information (1:forward strand 2:reverse strand)
  <li>splice junction information (0:noncanonical 1:GT/AG 2:CT/AC 3:GC/AG 4:CT/GC 5:AT/AC 6:GT/AT)
  <li>annotated? (1:true 0:false)
  <li>number of unique mapping reads
  <li>number of multimapping reads
  <li>maximum overhang
</ol>
Typically this file would have been generated by a previous run of the STAR aligner on the same genome and have a name <span class="filename">&lt;basename&gt;.SJ.out.tab</span>.
</ul>
You can provide both a GTF/GFF3 file and a tab file, the information they contain will be merged.

<h2>Output files</h2>
A STAR index consists of a series of files in a directory/folder. It is the name of this directory that is given as input parameter value to STAR for starting a mapping. For easy handling in GenePattern STAR.indexer puts this directory in a ZIP archive, which can be given as input to STAR.aligner.

<h2>Links</h2>
<ul>
<li>STAR at <a href="http://github.com/alexdobin/STAR"  target="_blank">GitHub</a>
</ul>

<h2>References</h2>
<ol>
<li>Dobin A, Davis CA, Schlesinger F, Drenkow J, Zaleski C, Jha S, Batut P, Chaisson M, Gingeras TR : STAR: ultrafast universal RNA-seq aligner. Bioinformatics 29(1):15-21 (2013). <a href="http://www.ncbi.nlm.nih.gov/pubmed/23104886" target="_blank">PubMed 23104886</a>
<li>Dobin A, Gingeras TR : Optimizing RNA-Seq Mapping with STAR. Methods Mol. Biol. 1415:245-265 (2016). <a href="http://www.ncbi.nlm.nih.gov/pubmed/27115637" target="_blank">PubMed 27115637</a>
</ol>

<h2>Author</h2>
The GenePattern interface is made by Guy Bottu, V.I.B.-B.I.T.S.
<p>
The STAR software is developed by a team of programmers headed by Alexander Dobin at Cold Spring Harbor Laboratory.

<h2>Version Comments</h2>
<table>
<tr><th>Version</th><th>Release date</th><th>Description</th><tr>
<tr><td>1</td><td>2016-07-07</td><td>for STAR 2.5.2a</td></tr>
</table>

</body>
</html>

